src/Mailbox.php

```diff
@@ -210,11 +210,14 @@ public function getIterator(): MessageIteratorInterface
      *
      * @param string $message
      *
-     * @return bool
+     * @return int sppended message UID
      */
-    public function addMessage(string $message): bool
+    public function addMessage(string $message): int
     {
-        return \imap_append($this->resource->getStream(), $this->getFullEncodedName(), $message);
+        $status = $this->getStatus(SA_UIDNEXT);
+        if (\imap_append($this->resource->getStream(), $this->getFullEncodedName(), $message)) {
+            return $status->uidnext;
+        }
```

Comment:
If `imap_append` returns false, the function would return null, and thus breaking the `int` return type.
Beside this tecnical issue, this change would give the user less informations than now.
Now the user can both call `getStatus` and have the `imap_append` return value; with this change the user would only have the message id, without another new functionality.
Last but not least, unfortunately this approach can lead to errors. Theoretically, between the `getStatus` call and the `imap_append` some other connection che use the `uidnext`, so you would return the wrong message id.